### ISSUE TITLE
3.13 support

### ISSUE THREAD

**Issue by raphaelauv** [sdispater/pendulum#844]
- [x] I have searched the [issues](https://github.com/sdispater/pendulum/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://pendulum.eustace.io/docs/) and believe that my question is not covered.

## Feature Request

Hi, an official support of python 3.13, thanks

**Comment by Secrus** [comment]
Is Pendulum incompatible right now?

**Comment by M5oul** [comment]
There are no Python 3.13 wheel provided. That’s the least I can say.
Releasing a new version on PyPI would provide a 3.13 wheel I guess.

**Comment by ericfrederich** [comment]
3.13 is released, hopefully a 3.13 wheel can be provided soon.  I don't want to install Rust to use Python ;-)

**Comment by ollz272** [comment]
@sdispater Is it possible to get this in, please?

**Comment by M5oul** [comment]
May be @Secrus?

**Comment by edgarrmondragon** [comment]
@ericfrederich @ollz272 @M5oul see https://github.com/sdispater/pendulum/issues/590#issuecomment-2405924505

**Comment by ericfrederich** [comment]
Some slightly bad news here.  While the tests are passing with Python 3.12 the tests do not pass on 3.13.

Python 3.13 [removed `uuid._load_system_functions`](https://github.com/python/cpython/pull/115934) which is called by `time-machine` which pendulum depends on.

Two ways of fixing this.

1: `poetry update time-machine` which updates time-machine from 2.13.0 -> 2.15.0 [where the call is gated based on sys.version_info](https://github.com/adamchainz/time-machine/pull/425).
2: bump minimum version to 3.9 which would update time-machine to 2.16.0 where they also dropped 3.8 support and [the call is removed](https://github.com/adamchainz/time-machine/pull/484)

**Comment by rudolfix** [comment]
tbh. it is hard to say why `time-machine` is in the dependencies. its only use is for tests, still this stuff is import into main `__init__`. or am I missing something?

**Comment by svrooij** [comment]
Are you still depending upon pendulum (the python library that requires rust to install), here is the PR how I removed the dependency upon it for Kiota.

It was quite easy since most functionality we were using is now supported by by Python itself. Actually the code we used is replaced by this code https://github.com/microsoft/kiota-python/blob/e9840dfd9acf75292e5c1cff3277a026a9a49140/packages/abstractions/kiota_abstractions/date_utils.py

https://github.com/microsoft/kiota-python/pull/450/files

**Comment by racinmat** [comment]
Are there any plans for dropping the python 3.8 support and updating, so it would passing on python 3.13?

**Comment by astr0gator** [comment]
👀

**Comment by racinmat** [comment]
Could we please get the new version released on pypi? @Secrus @ashb 
Thanks.

**Comment by Secrus** [comment]
@racinmat yes, the plan is to release it soon, we just want to review some things before the release.
